gateway/integration_test.go

```diff
@@ -325,3 +332,31 @@ func sendEvent(t *testing.T, httpPort int, payload *strings.Reader, callType, wr
 
 	t.Logf("Event Sent Successfully: (%s)", body)
 }
+
+func sendInternalBatch(t *testing.T, httpPort int, payload *strings.Reader, sourceID string) {
+	t.Helper()
+	t.Logf("Sending Internal Batch")
+
+	var (
+		httpClient = &http.Client{}
+		method     = "POST"
+		url        = fmt.Sprintf("http://localhost:%d/internal/v1/batch", httpPort)
+	)
+
+	req, err := http.NewRequest(method, url, payload)
+	require.NoError(t, err)
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Set("X-Rudder-Source-Id", sourceID)
+	req.Header.Add("AnonymousId", "anonymousId_header")
+
+	res, err := httpClient.Do(req)
```

Comment:
[optiona]: Using default client and some minor changes.
```suggestion
	batchURL := fmt.Sprintf("http://localhost:%d/internal/v1/batch", httpPort)

	req, err := http.NewRequest(http.MethodPost, batchURL, payload)
	require.NoError(t, err)

	req.Header.Add("Content-Type", "application/json")
	req.Header.Set("X-Rudder-Source-Id", sourceID)
	req.Header.Add("AnonymousId", "anonymousId_header")

	res, err := http.DefaultClient.Do(req)
```